plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/conversion/InlineMethodConverter.kt

```diff
@@ -5,26 +5,45 @@
 
 package org.jetbrains.kotlin.formver.conversion
 
+import org.jetbrains.kotlin.fir.expressions.FirLambdaArgumentExpression
 import org.jetbrains.kotlin.formver.embeddings.MethodSignatureEmbedding
 import org.jetbrains.kotlin.formver.embeddings.VariableEmbedding
 import org.jetbrains.kotlin.formver.viper.MangledName
 import org.jetbrains.kotlin.formver.viper.ast.Exp
 import org.jetbrains.kotlin.formver.viper.ast.Label
 
+sealed interface SubstitutionItem
+data class SubstitutionName(val name: MangledName) : SubstitutionItem
+data class SubstitutionLambda(val lambda: FirLambdaArgumentExpression) : SubstitutionItem
```

Comment:
Is this the level of abstraction that is convenient here?  Having the body already available here could be better.

Reply:
I am not sure that I am understanding what do you mean

Reply:
At this point you're storing the `FirLambdaArgumentExpression` as-is, but it seems like there may be things that are worth computing ahead of time and storing here: e.g. the body, perhaps something about the arguments?  One thing that comes to mind is that you could store a reference to the scope the lambda is defined in, so that you can know which variables not to mangle.